app/src/main/java/org/grammatek/simacorrect/spellcheckerservice/YfirlesturAnnotation.kt

```diff
@@ -35,105 +40,109 @@ class YfirlesturAnnotation(
+                    // if words are split up we have to fix the indexing from Yfirlestur
+                    // the method called tries to determine if that is the case.
+                    if ((annotation.start != annotation.end) && annotation.suggest != null) {
+                        annotationTokenEndIndex = adjustTokenIndices(annotation.text!!, annotation.start!!, annotation.end!!, i)
+                    }
+
+                    startChar = tokensIndices[i][annotationTokenStartIndex].startChar
+                    endChar = tokensIndices[i][annotationTokenEndIndex].endChar
+                    if (suggestions.size < suggestionsLimit && suggestion != null) {
+                        if (annotation.start == 0 && _unalteredOriginalText[startChar].isLowerCase()) {
+                            suggestion = suggestion.replaceFirstChar { it.lowercaseChar() }
+                        }
+                        suggestions.add(suggestion)
+                    }
                 }
-                endChar = annotation.endChar!!
-                if (suggestion.isNotEmpty() && annotation.suggest != null && suggestions.size <= suggestionsLimit) {
-                    suggestions.add(suggestion)
+                if(flags.isNotEmpty()) {
+                    val flag = flags.distinct().sum()
+                    suggestionList.add(SuggestionsInfo(flag, suggestions.toTypedArray()))
+                    suggestionsIndices.add(
+                        AnnotationIndices(startChar, endChar)
+                    )
                 }
             }
-            // We assign the cookie to 0 and re-assign it upstream where we have access to it.
-            suggestionList.add(SuggestionsInfo(flag, suggestions.toTypedArray(), 0, sequence))
-            suggestionsIndexes.add(
-                SuggestionIndexes(startChar, endChar)
-            )
         }
         return suggestionList
     }
 
     /**
-     * Finds the character sequence for given start and end index.
+     * Determines if an annotation has correct token indices
+     * and tries to correct them.
      *
-     * @param startChar
-     * @param endChar
-     * @return The character sequence hashed.
+     * @param text
+     * @param startIndex
+     * @param endIndex
+     * @return Int
      */
-    private fun getSequence(startChar: Int, endChar: Int): Int {
-        val sequenceString = _originalText.substring(startChar, endChar+1).trim()
-        return sequenceString.hashCode()
-    }
+    private fun adjustTokenIndices(text: String, startIndex: Int, endIndex: Int, sentenceIndex: Int): Int {
+        // Yfirlestur annotations have few cases where the token span is wrong.
+        // This only happens when Yfirlestur suggests a word be split into two.
+        // See https://github.com/mideind/Yfirlestur/issues/11 for more details.
+        val pattern = Regex("""'[^']*'""")
+        val realSuggestion = pattern.find(text)!!.value.replace("'", "")
```

Comment:
This can throw, if not pattern is found. How do you address that ?

Reply:
Interestingly enough, this method `adjustTokenIndices` doesn't appear to be necessary anymore.